src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/DateTimeSelection/OnBehalfDateTimeSelectionViewModel.swift

```diff
@@ -0,0 +1,170 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import Contacts
+import OpenCombine
+
+final class OnBehalfDateTimeSelectionViewModel {
+
+	// MARK: - Init
+
+	init(
+		traceLocation: TraceLocation,
+		onPrimaryButtonTap: @escaping (Checkin) -> Void
```

Comment:
I think there is no need to route this action through the view model. Could be resolved at the VC level.

Reply:
In this case it's actually useful as the view model assembles the Checkin and we can (and do in #3375) unit test that step if its in the model.